src/common/result.h

```diff
@@ -38,12 +33,16 @@ class Result
 public:
     Result(T value)
         : _result(std::move(value))
-        , _isError(false){};
-    Result(int code, QString str)
-        : _error({ std::move(str), code })
+        , _isError(false)
+    {
+    }
+    // TODO: This doesn't work if T and Error are too similar
```

Comment:
Yeah, the std::expected proposal includes a std::unexpected type for this purpose.
But other possibilities can be to have a  makeError(...)  function,  or a  Result::fromError

Does not have to be done in this change